components/header/Header.tsx

```diff
@@ -12,37 +13,91 @@ import {
   Paragraph,
 } from '@ost-cas-fee-adv-23-24/elbmum-design';
 import Image from 'next/image';
+import useUserInfo from '@/hooks/useUserInfo';
+import useModal from '@/hooks/useModal';
+import { EModalActions } from '@/stores/Modal.context';
+import ImagePreview from '@/components/image-preview/ImagePreview';
+import React from 'react';
+import ImageWithPlaceholder from '@/components/image-placeholder/ImageWithPlaceholder';
 
 interface Props {
   user: IPostCreator;
 }
 
 function Header({ user }: Props) {
+  const { identifier, setUserAvatar, avatarUrl, lastName, firstName } =
+    useUserInfo();
+  const { dispatchModal, closeModal } = useModal();
+  const imageSource = `https://source.unsplash.com/random/?landscape&${Date.now().toString().toLowerCase().trim()}`;
   return (
-    <div className="w-[680px]">
-      <div className="rounded-2xl overflow-hidden">
-        <Image
-          src={'/header.png'}
-          alt="Header Image"
-          width={680}
-          height={320}
-        />
-      </div>
-
-      <div className="absolute">
-        <Avatar
-          nameHtml="avatar"
-          size={EAvatarSizes.XL}
-          editable={false}
-          imgSrc={user.avatarUrl}
-        />
+    <>
+      <div className="relative">
+        <ImageWithPlaceholder src={imageSource} alt="header Image" />
+        <div className="absolute bottom-[-80px] right-8">
+          <Avatar
+            nameHtml="avatar"
+            size={EAvatarSizes.XL}
+            editable={user.id === identifier}
+            imgSrc={user.id === identifier ? avatarUrl : user.avatarUrl}
+            onSuccess={(newAvatar) => {
+              const reader = new FileReader();
+              reader.onload = (evt) => {
+                dispatchModal({
+                  type: EModalActions.SET_CONTENT,
+                  payload: {
+                    title: 'Lets change that ugly pic',
+                    content: (
+                      <>
+                        <ImagePreview
```

Comment:
Perhaps it would be clearer if we extract the content to its own component

Reply:
I created an issue for that: https://github.com/ost-cas-fee-adv-23-24/nextjs-app-football-fanatics/issues/30